File: backend/app/Models/Media.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Media extends Model
{
    protected $fillable = [
        'user_id', 'file_name', 'file_path',
        'file_type', 'file_size', 'disk', 'alt_text',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: backend/models/media.go
package models

type Media struct {
	BaseModel
	UserID   uint   `gorm:"not null" json:"user_id"`
	FileName string `gorm:"not null" json:"file_name"`
	FilePath string `gorm:"not null" json:"file_path"`
	FileSize int64  `json:"file_size"`
	MimeType string `json:"mime_type"`
	Disk     string `gorm:"default:'local'" json:"disk"`
	User     *User  `gorm:"foreignKey:UserID" json:"user,omitempty"`
}
